metsrw/premis/event.py

```diff
@@ -0,0 +1,262 @@
+
+        return cls(identifier_type, identifier_value, event_type, event_datetime, detail, outcomes, linking_agent_identifiers, linking_object_identifiers)
+
+    @classmethod
+    def _parse_outcomes(cls, outcome_information_els):
+        outcomes = []
+
+        for outcome_information_el in outcome_information_els:
+            details = []
+
+            detail_els = outcome_information_el.findall("premis:eventOutcomeDetail", namespaces=utils.NAMESPACES)
+
+            if detail_els is not None:
+                for detail_el in detail_els:
+                    extensions = detail_el.findall("premis:eventOutcomeDetailExtension", namespaces=utils.NAMESPACES)
+
+                    details.append({
+                        'note': detail_el.findtext("premis:eventOutcomeDetailNote", namespaces=utils.NAMESPACES),
+                        'extensions': extensions
+                    })
+
+            outcomes.append({
+                'outcome': outcome_information_el.findtext("premis:eventOutcome", namespaces=utils.NAMESPACES),
+                'details': details
+            })
+
+        return outcomes
+
+    @classmethod
+    def _parse_linking_agent_identifiers(cls, identifier_els):
+        identifiers = []
+
+        for identifier in identifier_els:
+            roles = []
+
+            for role_el in identifier.findall("premis:linkingAgentRole", namespaces=utils.NAMESPACES):
+                role = role_el.text if role_el is not None else None
+                if role is not None:
+                    roles.append(role)
+
+            identifiers.append({
+                'identifier_type': identifier.findtext("premis:linkingAgentIdentifierType", namespaces=utils.NAMESPACES),
+                'identifier_value': identifier.findtext("premis:linkingAgentIdentifierValue", namespaces=utils.NAMESPACES),
+                'roles': roles
+            })
+
+        return identifiers
+
+    @classmethod
+    def _parse_linking_object_identifiers(cls, identifier_els):
+        identifiers = []
+
+        for identifier in identifier_els:
+            roles = []
+
+            for role_el in identifier.findall("premis:linkingObjectRole", namespaces=utils.NAMESPACES):
+                role = role_el.text if role_el is not None else None
+                if role is not None:
+                    roles.append(role)
+
+            identifiers.append({
+                'identifier_type': identifier.findtext("premis:linkingObjectIdentifierType", namespaces=utils.NAMESPACES),
+                'identifier_value': identifier.findtext("premis:linkingObjectIdentifierValue", namespaces=utils.NAMESPACES),
+                'roles': roles
+            })
+
+        return identifiers
+
+    def serialize(self):
+        root = self._document_root()
+        root.append(self._serialize_identifier())
+
+        event_type_el = etree.Element(utils.lxmlns('premis') + 'eventType')
+        event_type_el.text = self.event_type
+        root.append(event_type_el)
```

Comment:
Tip: These three lines can be `etree.SubElement(root, utils.lxmlns('premis') + 'eventType').text = self.event_type`